test/asynchronous/test_encryption.py

```diff
@@ -2419,6 +2420,318 @@ async def test_05_roundtrip_encrypted_unindexed(self):
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        self.encrypted_client = await self.async_rs_or_single_client(
+            auto_encryption_opts=AutoEncryptionOpts(
+                key_vault_namespace="db.keyvault",
+                kms_providers={"local": {"key": LOCAL_MASTER_KEY}},
+            )
+        )
+        await self.encrypted_client.db.drop_collection("keyvault")
+
+        key_doc = json_data("etc", "data", "lookup", "key-doc.json")
+        key_vault = await create_key_vault(self.encrypted_client.db.keyvault, key_doc)
+        self.addCleanup(key_vault.drop)
+
+        await self.encrypted_client.db.drop_collection("csfle")
+        await self.encrypted_client.db.create_collection(
+            "csfle",
+            validator={"$jsonSchema": json_data("etc", "data", "lookup", "schema-csfle.json")},
+        )
+
+        await self.encrypted_client.db.drop_collection("csfle2")
+        await self.encrypted_client.db.create_collection(
+            "csfle2",
+            validator={"$jsonSchema": json_data("etc", "data", "lookup", "schema-csfle2.json")},
+        )
+
+        await self.encrypted_client.db.drop_collection("qe")
+        await self.encrypted_client.db.create_collection(
+            "qe", encryptedFields=json_data("etc", "data", "lookup", "schema-qe.json")
+        )
+
+        await self.encrypted_client.db.drop_collection("qe2")
+        await self.encrypted_client.db.create_collection(
+            "qe2", encryptedFields=json_data("etc", "data", "lookup", "schema-qe2.json")
+        )
+
+        await self.encrypted_client.db.drop_collection("no_schema")
+        await self.encrypted_client.db.create_collection("no_schema")
+
+        await self.encrypted_client.db.drop_collection("no_schema2")
+        await self.encrypted_client.db.create_collection("no_schema2")
+
+        self.unencrypted_client = await self.async_rs_or_single_client()
+
+        await self.encrypted_client.db.csfle.insert_one({"csfle": "csfle"})
+        doc = await self.unencrypted_client.db.csfle.find_one()
+        self.assertTrue(isinstance(doc["csfle"], Binary))
+        await self.encrypted_client.db.csfle2.insert_one({"csfle2": "csfle2"})
+        doc = await self.unencrypted_client.db.csfle2.find_one()
+        self.assertTrue(isinstance(doc["csfle2"], Binary))
+        await self.encrypted_client.db.qe.insert_one({"qe": "qe"})
+        doc = await self.unencrypted_client.db.qe.find_one()
+        self.assertTrue(isinstance(doc["qe"], Binary))
+        await self.encrypted_client.db.qe2.insert_one({"qe2": "qe2"})
+        doc = await self.unencrypted_client.db.qe2.find_one()
+        self.assertTrue(isinstance(doc["qe2"], Binary))
+        await self.encrypted_client.db.no_schema.insert_one({"no_schema": "no_schema"})
+        await self.encrypted_client.db.no_schema2.insert_one({"no_schema2": "no_schema2"})
+
+    @async_client_context.require_version_min(8, 1, -1)
+    async def test_1(self):
```

Comment:
For each test could you add a descriptive name, like `test_1_csfle_joins_no_schema`